test/jdk/java/foreign/TestSegmentAllocators.java

```diff
@@ -362,6 +378,20 @@ interface OfLongArray extends AllocationFunction<long[], ValueLayout.OfLong> { }
         interface OfDoubleArray extends AllocationFunction<double[], ValueLayout.OfDouble> { }
     }
 
+    record NamedAllocationFunction<X, L extends ValueLayout>(AllocationFunction<X, L> allocationFunction,
```

Comment:
If the goal of this is to show a string name in the test results, I think it can also be achieved by simply adding the string to the test parameters - e.g. 

```
new Object[] { factory,
                    ValueLayout.JAVA_DOUBLE.withOrder(ByteOrder.BIG_ENDIAN),
                    (AllocationFunction.OfDoubleArray) SegmentAllocator::allocateArray),
                    ToArrayHelper.toDoubleArray,
                    ""OfDoubleArray" // <----------------------
                    });
```

Which we also do in other tests